### DISCUSSION TITLE
what does paths-filter action do in pulsar-infra

### DISCUSSION THREAD

**Discussion by taojintianxia** [apache/pulsar#20639] [Q&A] [answered]
hello, pulsar team
I found there is an new action inin https://github.com/apache/pulsar/blob/master/.github/workflows/pulsar-ci.yaml
called paths-filter : `apache/pulsar-test-infra/paths-filter@master`

and I found this  action looks like the action `dorny/paths-filter`, did pulsar update anything on this ?
I do not understand why not use the `dorny/paths-filter` directly ?

**Comment by tisonkun** [comment, marked as answer]
> I do not understand why not use the dorny/paths-filter directly ?

ASF policy approves actions in the same org and `github` only. The project team can file an issue on ASF INFRA to whitelist some actions but sometime for simplicity we duplicate actions.
